### ISSUE TITLE
window.onerror [RECURSION] exception

### ISSUE THREAD

**Issue by ntalbott** [honeybadger-io/honeybadger-js#760]
## What are the steps to reproduce this issue?

1. Install the Honeybadger library to catch client-side JS errors.
2. Get lots of these errors.

## What happens?

<img width="1132" alt="Screen Shot 2022-04-22 at 1 09 31 PM" src="https://user-images.githubusercontent.com/223/164762996-80a44e2e-3f62-4328-a7f3-0d1c9fefbd5d.png">
<img width="1128" alt="Screen Shot 2022-04-22 at 1 10 09 PM" src="https://user-images.githubusercontent.com/223/164763026-d4e1a562-caa2-482d-a840-7bd8191ce4ea.png">

## What were you expecting to happen?

No error, or a useful error report.

## Any logs, error output, etc?

See above.

## Any other comments?
I think this in honeybadger.js since so far as I know I don't have any other onerror handlers installed.

## What versions are you using?
See screenshots.

**Comment by joshuap** [comment]
Can you take a look at this, @subzero10? I wonder if something is throwing an object that we're [flagging as recursive](https://github.com/honeybadger-io/honeybadger-js/blob/261668a5c675a172bd31ef21fb64e42d19013fd9/src/core/util.ts#L170).

**Comment by subzero10** [comment]
> Can you take a look at this, @subzero10? I wonder if something is throwing an object that we're [flagging as recursive](https://github.com/honeybadger-io/honeybadger-js/blob/261668a5c675a172bd31ef21fb64e42d19013fd9/src/core/util.ts#L170).

In order to flag one as recursive, it means that we are serializing an object that we've already seen (since the moment we call the `sanitize` method).
The best way to debug this would be a reproducible example.

cc @ntalbott @joshuap

**Comment by ntalbott** [comment]
@subzero10 I wish I could easily give you a reproduction, but this is a consumer product and this issue is happening enough to be more than noise but not enough for me to have ever seen it in the wild.

What's the best way to get the latest version of the JS library? Even hitting the url in the docs (`js.honeybadger.io/v3.2/honeybadger.min.js`) I'm getting 3.2.8, not 3.2.9.

**Comment by joshuap** [comment]
@ntalbott sorry about that—stale CDN cache, it seems. It should be 3.2.9 now.

**Comment by ntalbott** [comment]
OK, updated to 3.2.9, and even after burn-in, doesn't seem to have affected the rate of these (~100 hr, varied over time).

One question: with this getting thrown up to the HoneyBadger API, would an error message still get printed out to the console in the browser? Will try to keep an eye out if so and see if I can catch this in the wild.

**Comment by subzero10** [comment]
> One question: with this getting thrown up to the HoneyBadger API, would an error message still get printed out to the console in the browser? Will try to keep an eye out if so and see if I can catch this in the wild.

Yes an error message should be printed out to the console. You can also enable more verbose logging by setting `debug: true` when configuring Honeybadger.

**Comment by subzero10** [comment]
Hey @ntalbott, is this still an issue for you?

**Comment by ntalbott** [comment]
In the last week we’ve seen it 5-20 times/day, pretty much every day.


—
Nathaniel
<:((><
